ui/store/actions.ts

```diff
@@ -3282,6 +3282,45 @@ export function setParticipateInMetaMetrics(
   };
 }
 
+export function setDataCollectionForMarketing(
+  dataCollectionPreference: boolean,
+): ThunkAction<
+  Promise<[boolean, string]>,
+  MetaMaskReduxState,
+  unknown,
+  AnyAction
+> {
+  return (dispatch: MetaMaskReduxDispatch) => {
+    log.debug(`background.setDataCollectionForMarketing`);
+    return new Promise((resolve, reject) => {
+      callBackgroundMethod<string>(
```

Comment:
This function is deprecated, please use `submitRequestToBackground` instead

Reply:
Fixed